tensorflow/python/training/checkpoint_management.py

```diff
@@ -564,7 +564,27 @@ def __init__(self,
     active set may be deleted by this `CheckpointManager` or a future
     `CheckpointManager` instantiated in `directory` (subject to its
     `max_to_keep` and `keep_checkpoint_every_n_hours` settings).
-
+    
+    `CheckpointManager` can be used to save checkpoints every N steps.
+    An example usage is:
+    
+    ```python
+    # save checkpoints every 3000 steps and max_to_keep=3
+    # define your model
+    model = Model()
+    optimizer = tf.keras.optimizers.Adam()
+    checkpoint = tf.train.Checkpoint(model=model)
```

Comment:
Should the optimizer be checkpointed too? Otherwise restoring from a checkpoint will reset slot variables and start numbering at 0 again.

It's probably better to keep the first example as simple as possible. So maybe follow the example below and just have a blank Checkpoint, don't pass a step_counter argument, etc.? You'll also be able to do a doctested example in that case so we know the documentation isn't stale/broken.

Reply:
a demo in this link.
https://colab.research.google.com/drive/189p3S1ChHmi0lCswgwlnrjNiuor8DkLy?usp=sharing
you can run it.